### ISSUE TITLE
Support for variable length reads and variable insert sizes

### ISSUE THREAD

**Issue by jakebiesinger** [uci-cbcl/genomix#88]
For the scaffolding steps, we need to know how long each read is and its "outer distance", that is, the total distance between the two outside tips of the fragment.  We currently don't have any mechanism for recording this.

Reads belong to different "libraries", all of whose reads will have the same length and that will have the same insert size.  We currently don't have any way of indicating the library of a given read.

We need to record the _library-id_ in each read and then have _lookup tables_ that map library-id to read length and library-id to outer-distance.

**Comment by JavierJia** [comment]
Since we store the whole read sequence inside node,  can we just get the read length by `.length` ?

**Comment by jakebiesinger** [comment]
yep but that means we'd have to iterate over the whole collection when doing queries.  But maybe that wouldn't be such a bad approach if you're doing a batch process.

**Comment by jakebiesinger** [comment]
Plus, that doesn't help with the insert size problem, where the user specifies the value and it can be different depending on the library.
